### PR TITLE
[Merged by Bors] - feat(ZMod): Simple lemmas

### PR DESCRIPTION
**Pull request by YaelDillies**
From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Data/ZMod/Basic.lean [hunk 1]
```diff
@@ -1321,10 +1356,48 @@ theorem lift_comp_coe : ZMod.lift n f ∘ ((↑) : ℤ → _) = f :=
...
+lemma zmod_val_inv_nsmul_nsmul (hn : (Nat.card α).Coprime n) (a : α) :
+    (n⁻¹ : ZMod (Nat.card α)).val • n • a = a := by
+  rw [nsmul_left_comm, nsmul_zmod_val_inv_nsmul hn]
+
+end AddGroup
+
+section Group
+variable {α : Type*} [Group α] {n : ℕ}
+
+--TODO: Fix additivisation
```
Comment (riccardobrasca) [inline Mathlib/Data/ZMod/Basic.lean:1388]: Is this done?

File: Mathlib/Data/ZMod/Basic.lean [hunk 2]
```diff
@@ -1321,10 +1356,48 @@ theorem lift_comp_coe : ZMod.lift n f ∘ ((↑) : ℤ → _) = f :=
...
+    ZMod.intCast_surjective.forall, ZMod.lift_coe, AddMonoidHom.mem_ker, AddSubgroup.mem_bot]
+
 end lift
 
 end ZMod
 
+section AddGroup
+variable {α : Type*} [AddGroup α] {n : ℕ}
+
+--TODO: Additivise
```
Comment (riccardobrasca) [inline Mathlib/Data/ZMod/Basic.lean:1371]: What is the problem here?

File: Mathlib/Data/ZMod/Basic.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Data/ZMod/Basic.lean:1388]: No, sadly `to_additive` chokes on `Inv (ZMod n)`. cc @fpvandoorn

File: Mathlib/Data/ZMod/Basic.lean (hunk 2 above)
Comment (YaelDillies) [reply Mathlib/Data/ZMod/Basic.lean:1371]: Same as above 😢

File: Mathlib/Data/ZMod/Basic.lean (hunk 1 above)
Comment (riccardobrasca) [reply Mathlib/Data/ZMod/Basic.lean:1388]: OK, maybe add a (very) short explanation that this is the problem (of course if you want to add and issue and link it here even better...).

**Comment by riccardobrasca** [comment]
Thanks 🎉

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**